Src/FluentAssertions/Formatting/DefaultValueFormatter.cs

```diff
@@ -58,7 +59,7 @@ public void Format(object value, FormattedObjectGraph formattedGraph, Formatting
         /// <remarks>The default is all non-private members.</remarks>
         protected virtual MemberInfo[] GetMembers(Type type)
         {
-            return type.GetNonPrivateMembers().ToArray();
+            return type.GetNonPrivateMembers(MemberVisibility.Public).ToArray();
```

Comment:
Does this mean that if two objects only differ in an internal member, the formatted objects only contains the identical public members?

Reply:
Yep. Good point. Maybe we should include the internal `members` as well? Making it configurable seems a bit much to me.

Reply:
Without having any data to back me up, I think it might become too noisy for the default case if we include internal members.
E.g. for #889 one of the problems was that [HtmlNode](https://github.com/zzzprojects/html-agility-pack/blob/master/src/HtmlAgilityPack.Shared/HtmlNode.cs) has +20 internal backing fields.
Even with #899 when excluding already processed references, the formatted output has a lot of `avoiding circular references`.

If the failure message is clear enough about the missing/non-matching internal members, I would go with that for now and only print the public ones.